src/test/scala/circtTests/stage/ChiselStageSpec.scala

```diff
@@ -867,13 +867,21 @@ class ChiselStageSpec extends AnyFunSpec with Matchers with chiselTests.Utils {
         val a = IO(Input(Bool()))
         val b = IO(Output(Bool()))
 
+        // Chisel groups modules for deduping using their
```

Comment:
should there be a test about black boxes deduping (or not?) based on their desiredName? Is that what is going on?

Reply:
i added desiredNames to the hierarchy tests below and it's working, i think that should be enough?

Reply:
the hierarchy spec serves the same purpose since the instances are black boxed